logevents/src/test/java/org/logevents/observers/HumioLogEventObserverTest.java

```diff
@@ -0,0 +1,208 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import org.hamcrest.MatcherAssert;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.logevents.LogEvent;
+import org.logevents.extend.junit.LogEventSampler;
+import org.logevents.extend.junit.LogEventStatusRule;
+import org.logevents.formatting.MessageFormatter;
+import org.logevents.observers.batch.LogEventBatch;
+import org.logevents.status.LogEventStatus;
+import org.logevents.status.StatusEvent;
+import org.logevents.util.ExceptionUtil;
+import org.logevents.util.JsonUtil;
+
+public class HumioLogEventObserverTest {
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(9200);
+
+    private HumioLogEventObserver observer = new HumioLogEventObserver(toURL("http://localhost:9200"), "logevents-unit-test");
+
+    private URL toURL(String s) {
+        try {
+            return new URL(s);
+        } catch (MalformedURLException e) {
+            throw ExceptionUtil.softenException(e);
+        }
+    }
+
+    @Test
+    public void shouldFormatSimpleLogEvent() {
+        LogEvent event = new LogEventSampler().withMarker().build();
+        Map<String, Object> payload = observer.formatMessage(event);
+
+        assertEquals(event.getInstant(), ZonedDateTime.parse(payload.get("@timestamp").toString()).toInstant());
+        assertEquals(event.getLoggerName(), payload.get("logger"));
+        assertEquals(event.getLevel().name(), payload.get("level"));
+        assertEquals(event.getMessage(), payload.get("message"));
+        assertEquals(event.getThreadName(), payload.get("thread"));
+        assertEquals(event.getMarker().getName(), payload.get("marker"));
+    }
+
+    @Test
+    public void shouldWriteFormattedMessage() {
+        LogEvent event = new LogEventSampler().withArgs("a", "b", "c").build();
+        Map<String, Object> payload = observer.formatMessage(event);
+
+        assertEquals(event.getMessage(new MessageFormatter()), payload.get("message"));
+    }
+
+    @Test
+    public void shouldWriteMdcValues() {
+        LogEvent event = new LogEventSampler().withMdc("ip", "10.0.12.11").withMdc("op", "execute").build();
+        Map<String, Object> payload = observer.formatMessage(event);
+
+        assertEquals(event.getMdcProperties().get("ip"), JsonUtil.getObject(payload, "mdc").get("ip"));
+        assertEquals(event.getMdcProperties().get("op"), JsonUtil.getObject(payload, "mdc").get("op"));
+    }
+
+    @Test
+    public void shouldWriteExceptions() {
+        LogEvent event = new LogEventSampler().withThrowable().build();
+        Map<String, Object> payload = observer.formatMessage(event);
+
+        assertEquals(payload.get("exception.class"), event.getThrowable().getClass().getName());
+        assertEquals(payload.get("exception.message"), event.getThrowable().getMessage());
+        MatcherAssert.assertThat(payload.get("exception.stackTrace").toString(),
```

Comment:
As long as I'm not introducing AssertJ (a much better version of assertThat than hamcrest by the way) into Logevents, I'd prefer to duplicate a simple "assertContains" from here: https://github.com/jhannes/logevents/blob/main/logevents/src/test/java/org/logevents/observers/batch/SlackLogMessageFormatterTest.java#L163

Reply:
Fixed. Thanks for the link. Yes, I'm also a fan of AssertJ, but I simply stole existing code which already had MatcherAssert import in it from somewhere else in logevents :/